Add a function to display all segmented particles in one frame

YichiWang opened the pull request:
e.g.
particles = ps.ParticleAnalysis(image, params)
particles.show()

YichiWang committed:
test

test if local change can be commit to remote github

YichiWang committed:
Revert "test"

This reverts commit 6ec6344be89ab0b9c05f901c8b5cfe01d954924a.

YichiWang committed:
store segmented particles

Set default store_im to ture to store segmented particles after call ps.ParticleAnalysis

YichiWang committed:
add function to show all segmented particle images

eg. particles.show(). The displayed particles were normalize_boxing. Next step should enable option to display all particle images without normalize_boxing

TomSlater commented on ParticleSpy/ParticleAnalysis.py:
```diff
@@ -285,7 +285,7 @@ class parameters(object):
     
     def generate(self,threshold='otsu',watershed=False,watershed_size=50,
-                 watershed_erosion=5,invert=False,min_size=0,store_im=False,
+                 watershed_erosion=5,invert=False,min_size=0,store_im=True,
```
I'm not sure setting the default value to True is a good idea as some people might run in to memory issues. I'd prefer to keep it as False.

TomSlater commented:
Thanks Yichi, nice addition.

The PR should go to ePSIC-DLS/ParticleSpy rather than here.

I can add some documentation / tests too, unless you want to give it a go.

YichiWang committed:
pycache

YichiWang committed:
Set store_im to False

YichiWang committed:
pycache ignored

YichiWang committed:
Revert "pycache ignored"

This reverts commit 67ddc5b50c1403f69c3acefc2fcfafc26cc7ff27.

YichiWang committed:
Output particle image as a list of arrays

Allow segmented particle images output as a list of 2D numpy arrays (without hyperspy metadata).

e.g.
particles = ps.ParticleAnalysis(image, params)
particles.normalize_boxing()
pim_ls = particles.show(output=True)

YichiWang commented:
> Thanks Yichi, nice addition.
> 
> The PR should go to ePSIC-DLS/ParticleSpy rather than here.
> 
> I can add some documentation / tests too, unless you want to give it a go.

Hi Tom, perhaps you could add documentation / tests for this one to show me how it looks like so I can follow in the future. Thanks

YichiWang committed:
typo

YichiWang committed:
pycache ignore

YichiWang committed:
Revert "pycache ignore"

This reverts commit ed2a0937b417333f85807a531d7a09736af81e98.

YichiWang committed:
Revert "typo"

This reverts commit 19247096a93488e7c1271b68119859b8cf3523a5.

YichiWang committed:
pycache and parameter.h5 ignored

YichiWang committed:
ignore

YichiWang committed:
Allow show particle masks in a single frame

e.g.
particle_list.show('mask')
or particle_list.show('Image')

YichiWang committed:
adjust particle_list.show

Allow show segmented particles in a montage with adjustable number of columns.

e.g. particle_list.show(cols=1). Default montage try to display all particles into a square arrangement.

YichiWang committed:
Correct particle intensity_std calculation

Plus correct particle only image 'p_im'

YichiWang committed:
Allow user to select whether perform background subtraction before calculating image intensity

default is True, i.e. perform background subtraction

YichiWang committed:
create a new parameter "bkg_sub" and set default to True

Default is perform background subtraction for intensity calculation and for stored particle image.

YichiWang committed:
store background intensity into particle.properties

YichiWang committed:
Allow save no-paded particle image

With particle.store['pad']=None and particle.store['store_im']=True, the saved particle image is masked from original survey image and have save dimension as survey image and saved mask.

YichiWang committed:
correction

YichiWang committed:
Correct image intensity calculation

Main: Use np.ma.masked_array to mask particle image, so intensity std do not calculate the background area.

Others: 1. all intensity are background subtracted intensity. 2. Background value are saved as property.

YichiWang committed:
Create boundary_analysis.py

Few more functions. This is a standalone script and not merged with ParticleSpy yet.

-find boundary pixel in a clock-wise seqeunce
-calculate boundary curvature (signed) use least squre circle fit.
-average boundary curvature by given angle interval so at same direction boundary curvatures of two particle with different size (i.e. different boundary pixels) are comparable.

Normalised boundary curvature by particle's equivalent diameter is not implemented yet.

YichiWang committed:
Add reference for boundary following algorithm

YichiWang committed:
Revert "Add reference for boundary following algorithm"

This reverts commit e19dbd8aaeb1b96654a53e759df67a23474c85b1.

YichiWang committed:
Revert "Create boundary_analysis.py"

This reverts commit 1436108c780fcd53d4a049343476fd777b52c66a.
